src/Simplex/Messaging/Agent/Store/SQLite.hs

```diff
@@ -875,6 +1022,25 @@ getSndQueueByConnId_ dbConn connId =
        in Just SndQueue {server, sndId, sndPublicKey, sndPrivateKey, e2ePubKey, e2eDhSecret, status}
     sndQueue _ = Nothing
 
+getRcvQueueByPK_ :: DB.Connection -> RcvQueuePrimaryKey -> IO (Maybe RcvQueue)
+getRcvQueueByPK_ dbConn (SMPServer smpHost smpPort _, rcvQueueId) = do
```

Comment:
in what scenario it is needed? I think we always know connection ID, it seems a symptom of not passing it / storing it where it is needed?